src/theory/strings/theory_strings_rewriter.cpp

```diff
@@ -1440,6 +1444,15 @@ Node TheoryStringsRewriter::rewriteContains( Node node ) {
   Trace("strings-rewrite-debug2") << "No constant endpoints for " << node[0]
                                   << " " << node[1] << std::endl;
 
+  // length entailment
+  Node len_n1 = NodeManager::currentNM()->mkNode(kind::STRING_LENGTH, node[0]);
+  Node len_n2 = NodeManager::currentNM()->mkNode(kind::STRING_LENGTH, node[1]);
+  if (checkEntailArith(len_n2, len_n1, true))
```

Comment:
Just a thought: if non-strict `checkEntailArith` is true (and the strict version is false), then the two strings have to be exactly the same, i.e. we could rewrite the contains into an equality (not sure if creating equalities while rewriting makes any invariants fail though).

Reply:
Yes, this is possible and probably a good idea.  I'll add this.

Reply:
Notice we have to be careful about this however.  Technically 
checkEntailArith( v1, v2, true ) ---> false  (strict)
checkEntailArith( v1, v2, false ) ---> true (non-strict)
does not by the contract imply that v1 = v2.  The reason is because the entailment checks are incomplete.  The first statement simply says "We were unable to prove v1 > v2", it doesn't mean "v1 is never greater than v2".
So to be safe, I think we should actually use an equality entailment here, which would check that
Rewriter::rewrite( v1 - v2 ) = 0.

Reply:
Oh yes, this is a good point. We should probably address this in another PR in the future then?

Reply:
I went ahead and addressed it on this one, the entailment check for equality is fairly simple at this point (see checkEntailArithEq). This may be developed further later, but for now I've enabled this rewrite based on the simple check.

Reply:
This is great! Yes, there are probably some fairly easy checks that we can add in future PRs.